### PR TITLE
Add the charm to the model before setting the charm

### PR DESCRIPTION
**Pull request by hatched**
Juju 2.0 requires you to add the charm to the model before setting the service to that charm version. This adds that functionality to the inspectors change version component.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4474/
Test PASSed.

File: jujugui/static/gui/src/app/components/inspector-change-version/inspector-change-version.js [hunk 1]
```diff
@@ -68,13 +68,40 @@ YUI.add('inspector-change-version', function() {
...
     */
     _versionButtonAction: function(charmId, e) {
+      // Fetch the macaroon if one is available so that they can add private
+      // charms to their models.
+      var macaroon = this.props.getMacaroon();
+      // In Juju 2.0+ we have to add the charm before we set it. This is just
+      // good practice for 1.25.
+      this.props.addCharm(charmId, macaroon,
+        this._addCharmCallback.bind(this, charmId), {
+          // XXX hatch: the ecs doesn't yet support addCharm so we are going to
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/inspector-change-version/inspector-change-version.js]: Really? How do we react to addCharm bundle changeset messages?

File: jujugui/static/gui/src/app/components/inspector-change-version/inspector-change-version.js [hunk 2]
```diff
@@ -68,13 +68,40 @@ YUI.add('inspector-change-version', function() {
...
+          // send the command to juju immedaitely.
+          immediate: true
+        });
+    },
+
+    /**
+      Callback for handlign the results of addCharm
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/inspector-change-version/inspector-change-version.js]: typo: handling

Comment (frankban) [inline jujugui/static/gui/src/app/components/inspector-change-version/inspector-change-version.js]: ... and complete the sentence ;-)

File: jujugui/static/gui/src/app/components/inspector-change-version/test-inspector-change-version.js [hunk 3]
```diff
@@ -242,6 +242,8 @@ describe('InspectorChangeVersion', function() {
   it('can change charm version', function() {
     var changeState = sinon.stub();
     var serviceSet = sinon.stub();
+    var getMacaroon = sinon.stub().returns('macaroon');
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/inspector-change-version/test-inspector-change-version.js]: Cool, I need to take a look at this sinon library.

**Comment by frankban** [comment]
:+1: with a couple of minor requests.
Thank you!

File: jujugui/static/gui/src/app/components/inspector-change-version/inspector-change-version.js (hunk 2 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector-change-version/inspector-change-version.js]: oops, done.

File: jujugui/static/gui/src/app/components/inspector-change-version/test-inspector-change-version.js (hunk 3 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector-change-version/test-inspector-change-version.js]: http://sinonjs.org/ Yeah it is a really nice api for stubs and spys

File: jujugui/static/gui/src/app/components/inspector-change-version/inspector-change-version.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector-change-version/inspector-change-version.js]: We don't :) Basically we rely on the fakebackends implementation of loadCharm to fetch the charm data. https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/store/env/fakebackend.js#L329 so it only actually works as a 'side effect' which is why we've had to mark things as 'loaded' manually

**Comment by hatched** [comment]
Thanks for the review! :shipit: Will do the QA for the stack of branches that landed once this one is in there.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4476/
Test PASSed.

**Pull request merged**